File: LaboratoryWork/Semester_2/Laboratory_4/GoLang/task3/main.go
package main

import (
	"fmt"
	"time"
)

// XORShift32 — генератор псевдослучайных чисел
func xorshift32(seed *uint32) uint32 {
	*seed ^= *seed << 16
	*seed ^= *seed >> 21
	*seed ^= *seed << 11
	return *seed
}

func main() {
	// Используем текущее время в качестве начального значения
	seed := uint32(time.Now().Unix())

	for i := 0; i < 10; i++ {
		rnd := xorshift32(&seed)
		fmt.Printf("Случайное число: %d\n", rnd)
	}
}

File: LaboratoryWork/Semester_2/Laboratory_4/C++/task3/main.cpp
#include <iostream>
#include <cstdint>
#include <ctime>

using namespace std;

uint32_t xorshift32(uint32_t &seed) {
    // XORShift32
    seed ^= seed << 16;
    seed ^= seed >> 21;
    seed ^= seed << 11;
    return seed;
}

int main() {
    time_t t = time(0);
    uint32_t seed = t;

    for (int i = 0; i < 10; i++) {
        uint32_t rnd = xorshift32(seed);
        cout << "Случайное число: " << rnd << endl;
    }
}
